From 3e4fe73d51eb1adf8bb45cc00e66f0b2b5ee9bab Mon Sep 17 00:00:00 2001
From: Lans Zhang <jia.zhang@windriver.com>
Date: Wed, 21 Sep 2016 11:19:25 +0800
Subject: [PATCH 03/11] Allow to override the path to openssl

Upstream-Status: Pending

Signed-off-by: Lans Zhang <jia.zhang@windriver.com>
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 71d03aa..f63fe21 100644
--- a/Makefile
+++ b/Makefile
@@ -7,6 +7,7 @@ endif
 CC		= $(CROSS_COMPILE)gcc
 LD		= $(CROSS_COMPILE)ld
 OBJCOPY		= $(CROSS_COMPILE)objcopy
+OPENSSL		?= openssl
 
 ARCH		= $(shell $(CC) -dumpmachine | cut -f1 -d- | sed s,i[3456789]86,ia32,)
 OBJCOPY_GTE224  = $(shell expr `$(OBJCOPY) --version |grep ^"GNU objcopy" | sed 's/^.*\((.*)\|version\) //g' | cut -f1-2 -d.` \>= 2.24)
@@ -106,7 +107,7 @@ shim.crt:
 	./make-certs shim shim@xn--u4h.net all codesign 1.3.6.1.4.1.311.10.3.1 </dev/null
 
 shim.cer: shim.crt
-	openssl x509 -outform der -in $< -out $@
+	$(OPENSSL) x509 -outform der -in $< -out $@
 
 shim_cert.h: shim.cer
 	echo "static UINT8 shim_cert[] = {" > $@
-- 
2.11.0

