From a4f80f85079a4f52c94ec0dcc064b3ca26db919c Mon Sep 17 00:00:00 2001
From: Lans Zhang <jia.zhang@windriver.com>
Date: Wed, 21 Sep 2016 11:23:00 +0800
Subject: [PATCH 04/11] Fix for the cross compilation

Upstream-Status: Pending

Signed-off-by: Lans Zhang <jia.zhang@windriver.com>
---
 Makefile     | 12 ++++++++----
 lib/Makefile |  2 +-
 2 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index f63fe21..dcfa357 100644
--- a/Makefile
+++ b/Makefile
@@ -8,6 +8,10 @@ CC		= $(CROSS_COMPILE)gcc
 LD		= $(CROSS_COMPILE)ld
 OBJCOPY		= $(CROSS_COMPILE)objcopy
 OPENSSL		?= openssl
+HEXDUMP		?= hexdump
+PK12UTIL	?= pk12util
+CERTUTIL	?= certutil
+PESIGN		?= pesign
 
 ARCH		= $(shell $(CC) -dumpmachine | cut -f1 -d- | sed s,i[3456789]86,ia32,)
 OBJCOPY_GTE224  = $(shell expr `$(OBJCOPY) --version |grep ^"GNU objcopy" | sed 's/^.*\((.*)\|version\) //g' | cut -f1-2 -d.` \>= 2.24)
@@ -111,7 +115,7 @@ shim.cer: shim.crt
 
 shim_cert.h: shim.cer
 	echo "static UINT8 shim_cert[] = {" > $@
-	hexdump -v -e '1/1 "0x%02x, "' $< >> $@
+	$(HEXDUMP) -v -e '1/1 "0x%02x, "' $< >> $@
 	echo "};" >> $@
 
 version.c : version.c.in
@@ -122,8 +126,8 @@ version.c : version.c.in
 
 certdb/secmod.db: shim.crt
 	-mkdir certdb
-	pk12util -d certdb/ -i shim.p12 -W "" -K ""
-	certutil -d certdb/ -A -i shim.crt -n shim -t u
+	$(PK12UTIL) -d certdb/ -i shim.p12 -W "" -K ""
+	$(CERTUTIL) -d certdb/ -A -i shim.crt -n shim -t u
 
 shim.o: $(SOURCES) shim_cert.h
 shim.o: $(wildcard *.h)
@@ -185,7 +189,7 @@ endif
 		$(FORMAT) $^ $@.debug
 
 %.efi.signed: %.efi certdb/secmod.db
-	pesign -n certdb -i $< -c "shim" -s -o $@ -f
+	$(PESIGN) -n certdb -i $< -c "shim" -s -o $@ -f
 
 clean:
 	$(MAKE) -C Cryptlib clean
diff --git a/lib/Makefile b/lib/Makefile
index d93a26d..37f5746 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -5,7 +5,7 @@ LIBFILES = simple_file.o guid.o console.o execute.o configtable.o shell.o variab
 EFI_INCLUDES    = -I$(EFI_INCLUDE) -I$(EFI_INCLUDE)/$(ARCH) -I$(EFI_INCLUDE)/protocol -I../include
 
 lib.a: $(LIBFILES)
-	ar rcs lib.a $(LIBFILES)
+	$(AR) rcs lib.a $(LIBFILES)
 
 all: $(TARGET)
 
-- 
2.11.0

